src/community/oseo/oseo-core/src/main/java/org/geoserver/opensearch/eo/store/JDBCProductFeatureStore.java

```diff
@@ -101,6 +111,26 @@ && hasOutputProperty(query, OpenSearchAccess.COLLECTION_PROPERTY_NAME, false)) {
 
     @Override
     protected void mapPropertiesToComplex(ComplexFeatureBuilder builder, SimpleFeature fi) {
+        // JSONB Keys are Unsorted, so we sort them here
+        jsonBProperties.stream()
+                .filter(n -> fi.getAttribute(n) != null && fi.getAttribute(n) instanceof String)
+                .forEach(
+                        n -> {
+                            try {
+                                // convert from string to JSON
+                                JsonNode sortedJsonNode =
+                                        JSONFieldSupport.SORT_BY_KEY_MAPPER.readTree(
+                                                (String) fi.getAttribute(n));
+                                // convert back to string and set the attribute
+                                fi.setAttribute(n, sortedJsonNode.toString());
+                            } catch (JsonProcessingException e) {
+                                LOGGER.log(
+                                        java.util.logging.Level.WARNING,
+                                        "Error sorting JSONB field, could not parse JSON from field: "
+                                                + n,
+                                        e);
+                            }
```

Comment:
Create separate method to avoid excessive nesting?

Reply:
done